From cc64ec570c92fe39db3f1db8c877a8cc70e3b668 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Fri, 26 Oct 2012 21:53:31 +0100
Subject: [PATCH 047/278] avserver: use socklen_t where appropriate

Various socket functions expect a pointer to socklen_t which is not
necessarily int.

Signed-off-by: Mans Rullgard <mans@mansr.com>
---
 avserver.c |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/avserver.c b/avserver.c
index 994edcd..f9d85d8 100644
--- a/avserver.c
+++ b/avserver.c
@@ -799,7 +799,8 @@ static void http_send_too_busy_reply(int fd)
 static void new_connection(int server_fd, int is_rtsp)
 {
     struct sockaddr_in from_addr;
-    int fd, len;
+    socklen_t len;
+    int fd;
     HTTPContext *c = NULL;
 
     len = sizeof(from_addr);
@@ -1717,7 +1718,8 @@ static int http_parse_request(HTTPContext *c)
                     case REDIR_SDP:
                         {
                             uint8_t *sdp_data;
-                            int sdp_data_size, len;
+                            int sdp_data_size;
+                            socklen_t len;
                             struct sockaddr_in my_addr;
 
                             q += snprintf(q, c->buffer_size,
@@ -2991,7 +2993,8 @@ static void rtsp_cmd_describe(HTTPContext *c, const char *url)
     char path1[1024];
     const char *path;
     uint8_t *content;
-    int content_length, len;
+    int content_length;
+    socklen_t len;
     struct sockaddr_in my_addr;
 
     /* find which url is asked */
-- 
1.7.9.5

